File: day4-tool-system/tools/WeatherTool.ts
import { Tool } from "../Tool.js";
export class WeatherTool extends Tool {
    constructor() {
        super("getWeather", "Get the current weather for a given location.", {
            type: "object",
            properties: {
                location: {
                    type: "string",
                    description: "The location to get the weather for."
                }
            },
            required: ["location"]
        });
    }

    async execute( args: Record<string, unknown>): Promise<string> {
        const m: Record<string, string> = {
            北京: "晴，25°C，湿度 40%",
            上海: "多云，28°C，湿度 65%",
        };
        return m[args.location as string] ?? `${args.location}: 晴，22°C，湿度 50%`;

    }
}

File: day5-memory-system/tools/weather_tool.py
"""天气查询工具 — 模拟天气数据."""

from tool import Tool

MOCK_WEATHER = {
    "北京": "晴，25°C，湿度 40%，适合户外活动",
    "上海": "多云，28°C，湿度 65%",
    "广州": "雷阵雨，30°C，湿度 80%",
    "深圳": "阴，29°C，湿度 75%",
    "杭州": "小雨，22°C，湿度 70%",
    "成都": "阴，24°C，湿度 60%",
    "西安": "晴，27°C，湿度 35%",
    "武汉": "多云，26°C，湿度 55%",
}


class WeatherTool(Tool):
    def __init__(self):
        super().__init__(
            name="get_weather",
            description="获取指定城市的实时天气信息",
            parameters={
                "type": "object",
                "properties": {
                    "city": {"type": "string", "description": "城市名称"}
                },
                "required": ["city"],
            },
        )

    async def execute(self, args: dict) -> str:
        city = args.get("city", "")
        return MOCK_WEATHER.get(city, f"{city}: 晴，22°C，湿度 50%")
